src/lib/EventPublisher.js

```diff
@@ -61,12 +61,50 @@ class EventPublisher extends EventEmitter {
     this.emit(ALL_EVENTS, event);
   };
 
+  publishAndListenForResponse = async (eventData: EventData) => {
+    const eventID = uuid();
+    const requestEventName = `${eventData.name}/request`;
+    const responseEventName = `${eventData.name}/response/${eventID}`;
+
+    return new Promise((resolve, reject) => {
+      const responseListener = (event) => {
+        resolve(event);
+      };
+
+      this.subscribe(
+        responseEventName,
+        responseListener,
+        {} /* filterOptions */,
+        null /* subscriberID */,
+        {
+          once: true,
+          subscriptionTimeout: 5000,
+          timeoutHandler: () => reject(new Error('Timeout')),
```

Comment:
nit: add a better error for debugging purposes.  You should probably show the event name in there.